android/src/main/java/com/swmansion/rnscreens/ScreenStack.java

```diff
@@ -367,4 +408,81 @@ private void setupBackHandlerIfNeeded(ScreenStackFragment topScreen) {
       mFragmentManager.addOnBackStackChangedListener(mBackStackListener);
     }
   }
+
+  // below methods are taken from
+  // https://github.com/airbnb/native-navigation/blob/9cf50bf9b751b40778f473f3b19fcfe2c4d40599/lib/android/src/main/java/com/airbnb/android/react/navigation/ScreenCoordinatorLayout.java#L43
+  // and are used to swap the order of drawing views when navigating forward with the transitions
+  // that are making transitioning fragments appear one on another. See more info in the comment to
+  // the linked class.
+  @Override
+  public void removeView(final View view) {
+    if (isDetachingCurrentScreen) {
+      isDetachingCurrentScreen = false;
+      reverseLastTwoChildren = true;
+    }
+
+    super.removeView(view);
+  }
+
+  private void drawAndRelease() {
+    DrawingOp op = drawingOps.remove(0);
+    op.draw();
+    drawingOpPool.add(op);
+  }
+
+  @Override
+  protected void dispatchDraw(Canvas canvas) {
+    super.dispatchDraw(canvas);
+
+    // check the view removal is completed (by comparing the previous children count)
+    if (drawingOps.size() < previousChildrenCount) {
+      reverseLastTwoChildren = false;
+    }
+    previousChildrenCount = drawingOps.size();
+
+    if (reverseLastTwoChildren && drawingOps.size() >= 2) {
+      Collections.swap(drawingOps, drawingOps.size() - 1, drawingOps.size() - 2);
+    }
+
+    while (!drawingOps.isEmpty()) {
```

Comment:
i wonder why cant we just have a simple loop over items from drawingOps here. this isnt very crutial i believe, but still seems a bit obscure and strikes me when such iteration is done in n^2 while simple O(n) loop would suffice